workerapi/api_test.go

```diff
@@ -137,6 +149,47 @@ var tests = []struct {
 				StdoutRef: nonemptystr, StderrRef: nonemptystr, ExitCode: int(nonzero), Error: nonemptystr},
 		}},
 	},
+
+	// RunStatuses
+	{
+		15, rssFromThrift, nil,
+		[]*worker.RunStatus{
+			&worker.RunStatus{Status: worker.Status_COMPLETE, RunId: "a",
+				OutUri: nil, ErrUri: nil, Error: nil, ExitCode: &zero},
+			&worker.RunStatus{Status: worker.Status_COMPLETE, RunId: "b",
+				OutUri: nil, ErrUri: nil, Error: nil, ExitCode: &zero},
+		},
+		[]runner.RunStatus{
+			runner.RunStatus{RunID: "a", State: runner.COMPLETE,
+				StdoutRef: "", StderrRef: "", ExitCode: 0, Error: ""},
+			runner.RunStatus{RunID: "b", State: runner.COMPLETE,
+				StdoutRef: "", StderrRef: "", ExitCode: 0, Error: ""},
+		},
+	},
+
+	// RunQuery
```

Comment:
Looks like thrift struct RunsQuery fields are all optional - worth adding a test case with empty struct to make sure that works as intended (does nothing?)